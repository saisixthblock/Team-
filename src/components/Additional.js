import React, { Component } from "react";
import { Card,Row,Col, FormGroup, Label, Input } from "reactstrap";

class Additional extends Component{
    render(){
        return(
            <div>
                <Row>
                    <Col md={4}>
                        <FormGroup>
                            <Label htmlFor="location">Location:</Label>
                            <Input type="select" 
                                name="location"
                                placeholder="Select"
                                onChange={(e)=>this.props.handleChange(e.target.name, e.target.value)}
                                onBlur={(e)=>this.props.handleBlur(e.target.name,e.target.value)}
                                value={this.props.values.location}>
                                    <option value="">Select</option>
                                    <option value='Hyderabad'>Hyderabad</option>
                                    <option value='Chennai'>Chennai</option>
                                    <option value='Banglore'>Banglore</option>
                            </Input>
                            {this.props.values.locationError ? <p style={{color:'red'}}>{this.props.values.locationError}</p>: null}       
                        </FormGroup>
                    </Col>
                    <Col md={8}>
                        <FormGroup>
                            <Label htmlFor="gender">Gender:</Label><br/>
                            <Input type="radio"
                                name="gender"
                                value=""
                                checked={this.props.values.gender.male}
                                id="male" 
                                onChange={(e)=>this.props.handleChange(e.target.name, e.target.value)}
                                onBlur={(e)=>this.props.handleBlur(e.target.name,e.target.value)}
                            />&nbsp;
                            <Label htmlFor="male">male</Label>
                            <Input style={{marginLeft:5}} type="radio"
                                name="gender"
                                value=""
                                checked={this.props.values.gender.female}
                                id="female" 
                                onChange={(e)=>this.props.handleChange(e.target.name, e.target.value)}
                                onBlur={(e)=>this.props.handleBlur(e.target.name,e.target.value)}
                            />&nbsp;
                            <Label htmlFor="female">female</Label>
                            {this.props.values.genderError ? <p style={{color:'red'}}>{this.props.values.genderError}</p>: null}
                        </FormGroup>
                    </Col>
                </Row>
                <Row>
                    <Col>
                        <FormGroup>
                           <Label htmlFor="checkbox"> Time Slot :</Label><br/>
                           <Input type="checkbox" 
                                name="checkbox" 
                                id="morning"
                                value={this.props.values.checkbox.morning}
                                checked = {this.props.values.checkbox.morning}
                                onChange={(e)=>this.props.handleChange(e.target.name, e.target.value)}
                                onBlur={(e)=>this.props.handleBlur(e.target.name,e.target.value)}
                           />
                           <Label>Morning (9:00-12:00)</Label>
                           <Input style={{marginLeft:25}} 
                                type="checkbox" 
                                name="checkbox" 
                                id="evening"
                                value={this.props.values.checkbox.evening}
                                checked={this.props.values.checkbox.evening}
                                onChange={(e)=>this.props.handleChange(e.target.name, e.target.value)}
                                onBlur={(e)=>this.props.handleBlur(e.target.name,e.target.value)}
                            />
                            <Label>Evening (6:00-9:00)</Label>
                            {this.props.values.checkboxError ? <p style={{color:'red'}}>{this.props.values.checkboxError}</p>: null}
                        </FormGroup>
                    </Col>
                </Row>
            </div>
        )
    }
}
export default Additional