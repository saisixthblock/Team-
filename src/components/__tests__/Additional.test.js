import React from "react";
import Additional from "../Additional";
import Enzyme from 'enzyme';
import { shallow } from "enzyme";
import { shallowToJson } from "enzyme-to-json";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { Input } from "reactstrap";

Enzyme.configure({adapter:new Adapter() });

let sampleData={
    fullName:'',
    email:'abcd@gmail.com',
    password:'abcd',
    phoneNumber:'998877',
    address:'',
    city:'hyd',
    pincode:'502032',
    state:'',
    location:'Hyderabad',
    gender:"male",
    checkbox:'morning',
}

describe('Testing the Additional component',()=>{
    let wrapper=shallow(<Additional handleChange={jest.fn()} handleBlur={jest.fn()} values={sampleData}/>);
    it('Create the snapShot of Additional component', async ()=>{
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
    it('cheking the initaial values of component',()=>{
        expect(wrapper.find(Input).at(0).prop('value')).toEqual('Hyderabad');
        expect(wrapper.find(Input).at(1).prop('id')).toEqual('male');
        expect(wrapper.find(Input).at(2).prop('id')).toEqual('female');
        expect(wrapper.find(Input).at(3).prop('id')).toEqual('morning');
        expect(wrapper.find(Input).at(4).prop('id')).toEqual('evening');
    });
    it('Checking the handleChange for the additional component',()=>{
        wrapper.find(Input).at(0).prop('onChange')({target :{value :'Hyderabad'}});
        wrapper.find(Input).at(1).prop('onChange')({target :{value :'male'}});
        wrapper.find(Input).at(2).prop('onChange')({target :{value :'female'}});
        wrapper.find(Input).at(3).prop('onChange')({target :{value :'morning'}});
        wrapper.find(Input).at(4).prop('onChange')({target :{value :'evening'}});
    });
    it('Checking the handleBlur for the additional component',()=>{
        wrapper.find(Input).at(0).prop('onBlur')({target :{value :'Hyderabad'}});
        wrapper.find(Input).at(1).prop('onBlur')({target :{value :'male'}});
        wrapper.find(Input).at(2).prop('onBlur')({target :{value :'female'}});
        wrapper.find(Input).at(3).prop('onBlur')({target :{value :'morning'}});
        wrapper.find(Input).at(4).prop('onBlur')({target :{value :'evening'}});
    });
    it('Setting the props for the additional component',()=>{
        wrapper.setProps({
            ...sampleData,
            location:'Hyderabad',
            gender:'male',
            checkbox:'morning',
        });
    });
    it('cheking the values of component after setting props',()=>{
        expect(wrapper.find(Input).at(0).prop('value')).toEqual('Hyderabad');
        expect(wrapper.find(Input).at(1).prop('id')).toEqual('male');
        expect(wrapper.find(Input).at(2).prop('id')).toEqual('female');
        expect(wrapper.find(Input).at(3).prop('id')).toEqual('morning');
        expect(wrapper.find(Input).at(4).prop('id')).toEqual('evening');
    });

    
});
