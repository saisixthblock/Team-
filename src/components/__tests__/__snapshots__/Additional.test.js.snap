// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Additional component Create the snapShot of Additional component 1`] = `
<div>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      md={4}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <FormGroup
        tag="div"
      >
        <Label
          htmlFor="location"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
              "xxl",
            ]
          }
        >
          Location:
        </Label>
        <Input
          name="location"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Select"
          type="select"
          value="Hyderabad"
        >
          <option
            value=""
          >
            Select
          </option>
          <option
            value="Hyderabad"
          >
            Hyderabad
          </option>
          <option
            value="Chennai"
          >
            Chennai
          </option>
          <option
            value="Banglore"
          >
            Banglore
          </option>
        </Input>
      </FormGroup>
    </Col>
    <Col
      md={8}
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <FormGroup
        tag="div"
      >
        <Label
          htmlFor="gender"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
              "xxl",
            ]
          }
        >
          Gender:
        </Label>
        <br />
        <Input
          id="male"
          name="gender"
          onBlur={[Function]}
          onChange={[Function]}
          type="radio"
          value=""
        />
         
        <Label
          htmlFor="male"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
              "xxl",
            ]
          }
        >
          male
        </Label>
        <Input
          id="female"
          name="gender"
          onBlur={[Function]}
          onChange={[Function]}
          style={
            Object {
              "marginLeft": 5,
            }
          }
          type="radio"
          value=""
        />
         
        <Label
          htmlFor="female"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
              "xxl",
            ]
          }
        >
          female
        </Label>
      </FormGroup>
    </Col>
  </Row>
  <Row
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
        "xxl",
      ]
    }
  >
    <Col
      tag="div"
      widths={
        Array [
          "xs",
          "sm",
          "md",
          "lg",
          "xl",
          "xxl",
        ]
      }
    >
      <FormGroup
        tag="div"
      >
        <Label
          htmlFor="checkbox"
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
              "xxl",
            ]
          }
        >
           Time Slot :
        </Label>
        <br />
        <Input
          id="morning"
          name="checkbox"
          onBlur={[Function]}
          onChange={[Function]}
          type="checkbox"
        />
        <Label
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
              "xxl",
            ]
          }
        >
          Morning (9:00-12:00)
        </Label>
        <Input
          id="evening"
          name="checkbox"
          onBlur={[Function]}
          onChange={[Function]}
          style={
            Object {
              "marginLeft": 25,
            }
          }
          type="checkbox"
        />
        <Label
          tag="label"
          widths={
            Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
              "xxl",
            ]
          }
        >
          Evening (6:00-9:00)
        </Label>
      </FormGroup>
    </Col>
  </Row>
</div>
`;
